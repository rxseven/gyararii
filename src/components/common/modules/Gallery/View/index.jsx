import React, { useState } from 'react';
import { isMobile, isMobileOnly } from 'react-device-detect';
import styled from 'styled-components';

import Button from 'components/common/base/Button';
import Confirm from 'components/common/composite/Confirm';
import Toggle from 'components/common/composite/Toggle';
import Control from 'components/common/sections/Control';
import Error from 'components/common/composite/Error';
import File from 'components/common/composite/File';
import Lightbox from 'components/common/base/Lightbox';
import Jumper from 'components/common/base/Jumper';
import Masonry from 'components/common/sections/Masonry';
import Pagination from 'components/common/composite/Pagination';
import Toolbar from 'components/common/sections/Toolbar';

const Actions = styled.div`
  align-items: center;
  display: ${props => (props['data-visibility'] ? 'flex' : 'none')};
`;

const Autoscroll = styled.div`
  display: ${props => (props['data-visibility'] ? 'block' : 'none')};
`;

const Content = styled.div`
  align-items: center;
  display: flex;
  justify-content: ${({ narrow }) => (narrow ? 'flex-end' : 'space-between')};
`;

const Frame = styled.div`
  margin-top: ${({ isLoading }) => (isLoading ? '-1.5rem' : '36px')};
`;

function View(props) {
  const [isSelecting, setSelecting] = useState(false);
  const {
    action,
    autoscroll,
    currentImage,
    error,
    images,
    isDeleting,
    isFetching,
    isLightbox,
    isModal,
    isPreloading,
    isRestoring,
    isUploading,
    onAutoscroll,
    onDeleteCancel,
    onDeleteConfirm,
    onDeleteRequest,
    onDeselect,
    onDismiss,
    onLightboxClose,
    onLightboxNext,
    onLightboxOpen,
    onLightboxPrev,
    onLoad,
    onLoaded,
    onScrollTop,
    onSelect,
    onSelectAll,
    onUpload,
    pagination,
    selected,
    status
  } = props;
  const errorLength = error.length;
  const imageLength = images.length;
  const selectedLength = selected.length;
  const isLoading = isDeleting || isFetching || isUploading;
  const isMore = !!pagination;
  const isPrefetching =
    (isFetching && isPreloading) || (isRestoring && !imageLength);

  function handleSelecting() {
    setSelecting(!isSelecting);
  }

  return (
    <Frame isLoading={isPrefetching}>
      <Toolbar isLoading={isPrefetching}>
        <Content narrow={!!selectedLength}>
          <Actions data-visibility={!selectedLength}>
            <Button.Set>
              <File
                data-place="right"
                data-tip="Please DO NOT upload inappropriate images."
                data-tip-disable={isMobile}
                icon="cloud-upload-alt"
                id="gallery"
                isLoading={isLoading}
                multiple
                onUpload={onUpload}
                spinner={isUploading}
              >
                Upload
              </File>
              <If condition={isMobile}>
                <Button
                  icon="check"
                  isLoading={isLoading}
                  look={isSelecting ? 'primary' : 'secondary'}
                  onClick={handleSelecting}
                >
                  Tick
                </Button>
              </If>
              <Autoscroll data-visibility={!isMobileOnly}>
                <Toggle
                  checked={autoscroll}
                  isLoading={isLoading}
                  onToggle={onAutoscroll}
                >
                  Auto scroll
                </Toggle>
              </Autoscroll>
            </Button.Set>
          </Actions>
        </Content>
        <Control
          error={error}
          images={images}
          isLoading={isLoading}
          onDeleteRequest={onDeleteRequest}
          onDeselect={onDeselect}
          onError={onScrollTop}
          onSelectAll={onSelectAll}
          selected={selected}
          status={status}
        />
      </Toolbar>
      <Error error={error} onDismiss={onDismiss} />
      <Masonry
        isDeleting={isDeleting}
        isSelecting={isSelecting}
        error={errorLength}
        isFetching={isFetching}
        isMore={isMore}
        images={images}
        onPreview={onLightboxOpen}
        onLoaded={onLoaded}
        onSelect={onSelect}
        selected={selected}
        toDelete={action === 'delete'}
      />
      <Pagination
        isLoading={isFetching}
        isMore={!!imageLength && isMore}
        onLoad={onLoad}
      >
        More images
      </Pagination>
      <Confirm
        buttonConfirm="Delete"
        isOpen={isModal}
        isLoading={isDeleting}
        onClose={onDeleteCancel}
        onConfirm={onDeleteConfirm}
        title="Permanently Delete Images"
        type="danger"
      >
        <p>
          This action is permanent. Are you sure you donâ€™t want to reconsider?
        </p>
      </Confirm>
      <Lightbox
        currentImage={currentImage}
        images={images.map(image => ({ src: image.secure_url }))}
        isOpen={isLightbox}
        onClickNext={onLightboxNext}
        onClickPrev={onLightboxPrev}
        onClose={onLightboxClose}
      />
      <Jumper />
    </Frame>
  );
}

export default View;
