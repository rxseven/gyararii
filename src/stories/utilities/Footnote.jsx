import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
