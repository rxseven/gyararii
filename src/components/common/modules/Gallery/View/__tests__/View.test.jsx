import React from 'react';
import { factory } from 'tests/utilities';
import View from '../index';

// Arrange
const source = {
  // Common
  selected: [],

  // Toolbar
  nodeTop: React.createRef(),

  // File
  id: 'gallery',
  onUpload: jest.fn(),

  // Autoscroll
  isAutoscroll: false,
  onAutoscroll: jest.fn(),

  // Control
  onDeleteRequest: jest.fn(),
  onDeselect: jest.fn(),
  onScrollTop: jest.fn(),
  onSelectAll: jest.fn(),

  // Error
  error: [],
  onDismiss: jest.fn(),

  // Masonry
  images: [
    {
      public_id: 'image-001',
      secure_url: 'https://via.placeholder.com/480x320?text=Image-001',
      url: 'https://via.placeholder.com/480x320?text=Image-001'
    },
    {
      public_id: 'image-002',
      secure_url: 'https://via.placeholder.com/480x320?text=Image-002',
      url: 'https://via.placeholder.com/480x320?text=Image-002'
    },
    {
      public_id: 'image-003',
      secure_url: 'https://via.placeholder.com/480x320?text=Image-003',
      url: 'https://via.placeholder.com/480x320?text=Image-003'
    }
  ],
  onLightboxOpen: jest.fn(),
  onLoaded: jest.fn(),
  onSelect: jest.fn(),

  // Pagination
  isMore: false,
  onLoad: jest.fn(),

  // Confirm
  buttonConfirm: 'delete',
  isModal: false,
  onDeleteCancel: jest.fn(),
  onDeleteConfirm: jest.fn(),
  title: 'Permanently Delete Images',

  // Lightbox
  currentImage: 1,
  onLightboxNext: jest.fn(),
  onLightboxPrev: jest.fn(),
  onLightboxClose: jest.fn()
};

// Setup
function setup(props) {
  return factory(View, source, props);
}

// Test suites
describe('<Gallery.View />', () => {
  it('should render without crashing', () => {
    setup();
  });
});
