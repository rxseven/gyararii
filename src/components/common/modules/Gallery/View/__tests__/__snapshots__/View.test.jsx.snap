// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery.View /> Snapshot tests should render correctly 1`] = `
<div
  className="sc-kEYyzF hhZbae"
>
  <div
    className="sc-eNQAEJ iJLkrf"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <div
            className="sc-jKJlTe iXSmQl"
          >
            <label
              className="sc-gqjmRU egfDhJ"
              htmlFor="gallery"
            >
              <span
                className="btn btn-primary btn-sm"
                data-place="right"
                data-tip="Please DO NOT upload inappropriate images."
                disabled={false}
                role="button"
              >
                <span
                  className="sc-htpNat iYntvg"
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-cloud-upload-alt fa-w-20 "
                    data-icon="cloud-upload-alt"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 640 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4zM393.4 288H328v112c0 8.8-7.2 16-16 16h-48c-8.8 0-16-7.2-16-16V288h-65.4c-14.3 0-21.4-17.2-11.3-27.3l105.4-105.4c6.2-6.2 16.4-6.2 22.6 0l105.4 105.4c10.1 10.1 2.9 27.3-11.3 27.3z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </span>
                <span
                  className="sc-bxivhb iGoLHq"
                >
                  Upload
                </span>
                <span
                  className="sc-bwzfXH iDiJqN"
                >
                  <span>
                    <style>
                       @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
                    </style>
                    <div
                      style={
                        Object {
                          "WebkitAnimationDelay": "-0.32s",
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDelay": "-0.32s",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "marginRight": 5,
                          "width": 7,
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "WebkitAnimationDelay": "-0.16s",
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDelay": "-0.16s",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "marginRight": 5,
                          "width": 7,
                        }
                      }
                    />
                    <div
                      style={
                        Object {
                          "WebkitAnimationDuration": "1.25s",
                          "WebkitAnimationIterationCount": "infinite",
                          "WebkitAnimationName": "brsk-three-bounce",
                          "WebkitAnimationTimingFunction": "ease-in-out",
                          "animationDuration": "1.25s",
                          "animationIterationCount": "infinite",
                          "animationName": "brsk-three-bounce",
                          "animationTimingFunction": "ease-in-out",
                          "backgroundColor": "#fff",
                          "borderRadius": "100%",
                          "display": "inline-block",
                          "height": 7,
                          "width": 7,
                        }
                      }
                    />
                  </span>
                </span>
              </span>
              <input
                className="sc-gZMcBi jOngK"
                disabled={false}
                id="gallery"
                multiple={true}
                onChange={[MockFunction]}
                type="file"
              />
            </label>
            <div
              className="sc-gzVnrw liWgEh"
            >
              <div
                className="sc-dnqmqq guwrEu"
              >
                Showing 
                <span
                  className="sc-EHOje lpjwMY"
                >
                  3
                </span>
                 images
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-hMqMXs dRzBD"
      />
    </div>
  </div>
  <div
    className="sc-kGXeez jhzAZZ"
  >
    <div
      className="card-columns"
    >
      <div
        className="card sc-kgoBCf dEuInR text-text-left mb-3"
      >
        <div
          className="sc-fjdhpX idBwjd"
          onClick={[Function]}
        >
          <div
            className="sc-chPdSV gOGbZh"
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check-circle fa-w-16 sc-kAzzGY jQRtnk"
            data-checked={false}
            data-icon="check-circle"
            data-mobile={false}
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            className="sc-jzJRlG fSxHWO"
            data-mobile={false}
            href="https://via.placeholder.com/480x320?text=Image-001"
            onClick={[Function]}
            target="_blank"
            title="Open image in new tab"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-external-link-alt fa-w-18 "
              data-icon="external-link-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <div
            className="sc-cSHVUG bHUcXy"
          >
            <span>
              <style>
                 @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
              </style>
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.32s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.32s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.16s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.16s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "width": 7,
                  }
                }
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="card sc-kgoBCf dEuInR text-text-left mb-3"
      >
        <div
          className="sc-fjdhpX idBwjd"
          onClick={[Function]}
        >
          <div
            className="sc-chPdSV gOGbZh"
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check-circle fa-w-16 sc-kAzzGY jQRtnk"
            data-checked={false}
            data-icon="check-circle"
            data-mobile={false}
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            className="sc-jzJRlG fSxHWO"
            data-mobile={false}
            href="https://via.placeholder.com/480x320?text=Image-002"
            onClick={[Function]}
            target="_blank"
            title="Open image in new tab"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-external-link-alt fa-w-18 "
              data-icon="external-link-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <div
            className="sc-cSHVUG bHUcXy"
          >
            <span>
              <style>
                 @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
              </style>
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.32s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.32s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.16s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.16s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "width": 7,
                  }
                }
              />
            </span>
          </div>
        </div>
      </div>
      <div
        className="card sc-kgoBCf dEuInR text-text-left mb-3"
      >
        <div
          className="sc-fjdhpX idBwjd"
          onClick={[Function]}
        >
          <div
            className="sc-chPdSV gOGbZh"
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check-circle fa-w-16 sc-kAzzGY jQRtnk"
            data-checked={false}
            data-icon="check-circle"
            data-mobile={false}
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <a
            className="sc-jzJRlG fSxHWO"
            data-mobile={false}
            href="https://via.placeholder.com/480x320?text=Image-003"
            onClick={[Function]}
            target="_blank"
            title="Open image in new tab"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-external-link-alt fa-w-18 "
              data-icon="external-link-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <div
            className="sc-cSHVUG bHUcXy"
          >
            <span>
              <style>
                 @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
              </style>
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.32s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.32s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDelay": "-0.16s",
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDelay": "-0.16s",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "marginRight": 5,
                    "width": 7,
                  }
                }
              />
              <div
                style={
                  Object {
                    "WebkitAnimationDuration": "1.25s",
                    "WebkitAnimationIterationCount": "infinite",
                    "WebkitAnimationName": "brsk-three-bounce",
                    "WebkitAnimationTimingFunction": "ease-in-out",
                    "animationDuration": "1.25s",
                    "animationIterationCount": "infinite",
                    "animationName": "brsk-three-bounce",
                    "animationTimingFunction": "ease-in-out",
                    "backgroundColor": "#fff",
                    "borderRadius": "100%",
                    "display": "inline-block",
                    "height": 7,
                    "width": 7,
                  }
                }
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-ckVGcZ cAgsBG"
    hidden={true}
  >
    <span>
      <style>
         @-webkit-keyframes brsk-three-bounce { 0%, 80%, 100% { -webkit-transform: scale(0); } 40% { -webkit-transform: scale(1); } } @keyframes brsk-three-bounce { 0%, 80%, 100% { transform: scale(0); -webkit-transform: scale(0); } 40% { transform: scale(1); -webkit-transform: scale(1); } } 
      </style>
      <div
        style={
          Object {
            "WebkitAnimationDelay": "-0.32s",
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDelay": "-0.32s",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "marginRight": 5,
            "width": 7,
          }
        }
      />
      <div
        style={
          Object {
            "WebkitAnimationDelay": "-0.16s",
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDelay": "-0.16s",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "marginRight": 5,
            "width": 7,
          }
        }
      />
      <div
        style={
          Object {
            "WebkitAnimationDuration": "1.25s",
            "WebkitAnimationIterationCount": "infinite",
            "WebkitAnimationName": "brsk-three-bounce",
            "WebkitAnimationTimingFunction": "ease-in-out",
            "animationDuration": "1.25s",
            "animationIterationCount": "infinite",
            "animationName": "brsk-three-bounce",
            "animationTimingFunction": "ease-in-out",
            "backgroundColor": "#999",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": 7,
            "width": 7,
          }
        }
      />
    </span>
  </div>
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": 0,
        "cursor": "pointer",
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "transitionDelay": "0s",
        "transitionDuration": "0.2s",
        "transitionProperty": "opacity, visibility",
        "transitionTimingFunction": "linear",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="sc-jTzLTM iMLVic"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-up fa-w-14 sc-VigVT iSMghy"
        data-icon="arrow-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.9 289.5l-22.2-22.2c-9.4-9.4-9.4-24.6 0-33.9L207 39c9.4-9.4 24.6-9.4 33.9 0l194.3 194.3c9.4 9.4 9.4 24.6 0 33.9L413 289.4c-9.5 9.5-25 9.3-34.3-.4L264 168.6V456c0 13.3-10.7 24-24 24h-32c-13.3 0-24-10.7-24-24V168.6L69.2 289.1c-9.3 9.8-24.8 10-34.3.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
